### PR TITLE
SLING-8409 - Improve test coverage

### PR DESCRIPTION
**Pull request by cschneider**

### CODE REVIEW

File: src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java [hunk 1]
```diff
@@ -150,37 +152,45 @@ public Closeable createPoller(String topicName, Reset reset, @Nullable String as
...
+                ? consumer.beginningOffsets(topicPartitions)
+                : consumer.endOffsets(topicPartitions);
     }
 
     @Override
     public String assignTo(long offset) {
         return format("%s:%s", PARTITION, offset);
     }
 
+    protected <T> KafkaConsumer<String, T> createConsumer(Class<? extends Deserializer<?>> deserializer, Reset reset) {
```
Comment (cschneider) [inline src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java]: I used a method to create the consumer so I can override it for the tests.

File: src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java [hunk 2]
```diff
@@ -150,37 +152,45 @@ public Closeable createPoller(String topicName, Reset reset, @Nullable String as
...
-        String groupId = UUID.randomUUID().toString();
-        KafkaConsumer<String, byte[]> consumer = new KafkaConsumer<>(consumerConfig(ByteArrayDeserializer.class, groupId, reset));
-        TopicPartition topicPartition = new TopicPartition(topicName, PARTITION);
-        Collection<TopicPartition> topicPartitions = singleton(topicPartition);
-        final Map<TopicPartition, Long> offsets;
-        if (reset == Reset.earliest) {
-            offsets = consumer.beginningOffsets(topicPartitions);
-        } else {
-            offsets = consumer.endOffsets(topicPartitions);
+        try (KafkaConsumer<String, String> consumer = createConsumer(StringDeserializer.class, reset)) {;
```
Comment (cschneider) [inline src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java]: This is a pure refactoring

File: src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java [hunk 3]
```diff
@@ -150,37 +152,45 @@ public Closeable createPoller(String topicName, Reset reset, @Nullable String as
...
-            if (! consumer.listTopics().containsKey(topic)) {
-                throw new MessagingException(format("Topic %s does not exist", topic));
-            }
+        Map<String, List<PartitionInfo>> topics;
+        try (KafkaConsumer<String, String> consumer = createConsumer(StringDeserializer.class, Reset.latest)) {
+            topics = consumer.listTopics();
         } catch (Exception e) {
             throw new MessagingException(format("Unable to load topic stats for %s", topic), e);
         }
+        if (! topics.containsKey(topic)) {
```
Comment (cschneider) [inline src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java]: Moving this outside the try catch avoids rewrapping MessageException

File: src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java [hunk 4]
```diff
@@ -112,7 +115,7 @@ public void close() {
     @Override
     public Closeable createPoller(String topicName, Reset reset, @Nullable String assign, HandlerAdapter<?>... adapters) {
         String consumerGroupId = UUID.randomUUID().toString();
-        KafkaConsumer<String, byte[]> consumer = new KafkaConsumer<>(consumerConfig(ByteArrayDeserializer.class, consumerGroupId, reset));
+        KafkaConsumer<String, byte[]> consumer = createConsumer(ByteArrayDeserializer.class, reset);
```
Comment (tmaret) [inline src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java]: Due to this refactoring the `info` level log statement below is wrong (the `consumerGroupId` is not the one used in the message poller).

Comment (cschneider) [reply src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java]: Thanks .. will fix it

Comment (tmaret) [reply src/main/java/org/apache/sling/distribution/journal/kafka/KafkaClientProvider.java]: thanks @cschneider !

**Pull request merged**